### PR TITLE
Cut and paste modifications

### PR DESCRIPTION
**Pull request by archendel**
Signed-off-by: LE SAULNIER Kevin <kevin.lesaulnier@rte-france.com>

### CODE REVIEW

**Review by flomillot — APPROVED** [review]

File: src/utils/rest-api.js [hunk 1]
```diff
@@ -1165,6 +1166,35 @@ export function deleteModifications(studyUuid, nodeUuid, modificationUuid) {
     );
 }
 
+export function cutAndPasteModifications(
```
[the remark this answers is not in the document]
Comment (archendel) [reply src/utils/rest-api.js]: fixed

File: src/utils/rest-api.js [hunk 2]
```diff
@@ -7,6 +7,7 @@
 import { store } from '../redux/store';
 import ReconnectingWebSocket from 'reconnecting-websocket';
 import { APP_NAME, getAppName } from './config-params';
+import { CopyType } from '../components/graph/menus/network-modification-node-editor';
```
[the remark this answers is not in the document]
Comment (archendel) [reply src/utils/rest-api.js]: removed with the previous fix

**Review by sBouzols — CHANGES_REQUESTED** [review]
missing case :
for a public study openned with to user.
user1, user2 showing the same node.
user1 cut modifications in node and move it to another node
user2 isn't notified of this move and still see th node with modifications that has been moved by user1

File: src/components/graph/menus/network-modification-node-editor.js [hunk 3]
```diff
@@ -501,27 +501,30 @@ const NetworkModificationNodeEditor = () => {
...
                 copiedModifications,
-                copyType
+                copyInfos
             )
                 .then(() => {
-                    setCopyType(null);
+                    setCopyInfos(null);
                     setCopiedModifications([]);
                 })
                 .catch((errmsg) => {
```
Comment (sBouzols) [inline src/components/graph/menus/network-modification-node-editor.js:530]: maybe add a `errMoveModificationMsg`

File: src/utils/rest-api.js [hunk 4]
```diff
@@ -1169,17 +1169,20 @@ export function copyOrMoveModifications(
     studyUuid,
     targetNodeId,
     modificationToCutUuidList,
-    copyType
+    copyInfos
 ) {
-    console.info(copyType + ' modifications');
+    console.info(copyInfos.copyType + ' modifications');
     const cutAndPasteModificationUrl =
```
Comment (sBouzols) [inline src/utils/rest-api.js:1175]: rename `cutAndPasteModificationUrl` to `copyOrMoveModificationUrl`

File: src/components/graph/menus/network-modification-node-editor.js (hunk 3 above)
Comment (archendel) [reply src/components/graph/menus/network-modification-node-editor.js:530]: fixed

File: src/utils/rest-api.js (hunk 4 above)
Comment (archendel) [reply src/utils/rest-api.js:1175]: fixed

**Review by sBouzols — APPROVED** [review]

**Pull request merged**